libraries/plugins/peerplays_sidechain/peerplays_sidechain_plugin.cpp

```diff
@@ -343,6 +345,40 @@ void peerplays_sidechain_plugin_impl::recreate_primary_wallet()
 }
 
 void peerplays_sidechain_plugin_impl::process_deposits() {
+
+   // Account who issues tokens to the user who made deposit
+   account_id_type pay_from = GRAPHENE_NULL_ACCOUNT;
+   const auto& account_idx = plugin.database().get_index_type<account_index>().indices().get<by_name>();
+   const auto& account_itr = account_idx.find("nathan");
+   if (account_itr != account_idx.end())
+      pay_from = (*account_itr).id;
+
+   const auto& idx = plugin.database().get_index_type<son_wallet_transfer_index>().indices().get<by_processed>();
+   const auto& idx_range = idx.equal_range(false);
+
+   std::for_each(idx_range.first, idx_range.second,
+         [&] (const son_wallet_transfer_object& swto) {
+
+      const chain::global_property_object& gpo = plugin.database().get_global_properties();
+
```

Comment:
I dont think so. This part is excuted according to SON scheduling, SONS will, one by one, look into this list, and process what they find there.

Reply:
What I meant was the following scenario,

- son1 is scheduled next.
- son1 detects an unprocessed transfer object.
- son1 sends a proposal p1.
- the proposal p1 is not approved yet ( swto.processed is not true yet )
- son2 is scheduled next.
- son2 also detects an unprocessed transfer object.
- son1 sends a proposal p2.

Reply:
Yes, and this can not be avoided. My solution for this is that operation.evaluate() checks whether the son_wallet_transfer_object is processed or not. Then, it can silently skip to do what it is supposed to do, or fail to execute. I choosed first approach, because proposal with operation that fails will retry to execute on each new approval and proposal expiration.

Reply:
We can look into this later but before going into production we will need to add a check to find any active proposals already in review. Otherwise, SON PPYs are burnt unnecessarily for proposal creation operation.